simulator/examples/clock.rs

```diff
@@ -0,0 +1,196 @@
+            .stroke(Some(BinaryColor::On))
+            .into_iter()
+    });
+
+    // Create a single iterator of pixels, first iterating over the circle, then over the 12 lines
+    // generated
+    face.into_iter().chain(tics.flatten())
+}
+
+/// Draw the seconds hand given a seconds value (0 - 59)
+fn draw_seconds_hand(seconds: u32) -> impl Iterator<Item = Pixel<BinaryColor>> {
+    // Convert seconds into a position around the circle in radians
+    let seconds_radians = ((seconds as f32 / 60.0) * 2.0 * PI) + START;
+
+    let end = CENTER
+        + Point::new(
+            (seconds_radians.cos() * (SIZE as f32)) as i32,
+            (seconds_radians.sin() * (SIZE as f32)) as i32,
+        );
+
+    // Basic line hand
+    let hand = Line::new(CENTER, end).stroke(Some(BinaryColor::On));
+
+    // Offset from end of hand
+    let decoration_offset = Point::new(
+        (seconds_radians.cos() * (20.0)) as i32,
+        (seconds_radians.sin() * (20.0)) as i32,
+    );
+
+    // Add a fancy circle near the end of the hand
+    let decoration = Circle::new(end - decoration_offset, 5)
+        .fill(Some(BinaryColor::Off))
+        .stroke(Some(BinaryColor::On))
+        .stroke_width(1);
+
+    hand.into_iter().chain(decoration)
+}
+
+/// Draw the hour hand (0-11)
+fn draw_hour_hand(hour: u32) -> Line<BinaryColor> {
+    // Convert hour into a position around the circle in radians
+    let hour_radians = ((hour as f32 / 12.0) * 2.0 * PI) + START;
+
+    let hand_len = SIZE as f32 - 60.0;
+
+    let end = CENTER
+        + Point::new(
+            (hour_radians.cos() * hand_len) as i32,
+            (hour_radians.sin() * hand_len) as i32,
+        );
+
+    // Basic line hand
+    Line::new(CENTER, end).stroke(Some(BinaryColor::On))
+}
+
+/// Draw the minute hand (0-59)
+fn draw_minute_hand(minute: u32) -> Line<BinaryColor> {
+    // Convert minute into a position around the circle in radians
+    let minute_radians = ((minute as f32 / 60.0) * 2.0 * PI) + START;
+
+    let hand_len = SIZE as f32 - 30.0;
+
+    let end = CENTER
+        + Point::new(
+            (minute_radians.cos() * hand_len) as i32,
+            (minute_radians.sin() * hand_len) as i32,
+        );
+
+    // Basic line hand
+    Line::new(CENTER, end).stroke(Some(BinaryColor::On))
+}
+
+/// Draw digital clock just above center with black text on a white background
+///
+/// NOTE: The formatted time str must be passed in as references to temporary values in a
+/// function can't be returned.
+fn draw_digital_clock<'a>(time_str: &'a str) -> impl Iterator<Item = Pixel<BinaryColor>> + 'a {
+    let text = Font12x16::<BinaryColor>::render_str(&time_str)
+        .stroke(Some(BinaryColor::Off))
+        .translate(CENTER - Size::new(48, 48));
+
+    // Add some padding on the background around the time digits, here it's 2px in each direction
+    let offset = Size::new(2, 2);
+
+    let background = Rectangle::new(text.top_left() - offset, text.bottom_right() + offset)
```

Comment:
Because the font rendering already adds some padding to the bottom right this looks better if the offset is only applied to the top left.